docs/tutorials/basics/basics_backend.md

```diff
@@ -0,0 +1,99 @@
+launcher script for MOBSTER (promotion-type), using the SageMaker instead of
+the local back-end.
+* [1] The only difference is that `backend`, feeding into the `Tuner`, is
+  different. Instead of the code `entry_point` only, you need to specify a
+  [SageMaker estimator](https://sagemaker.readthedocs.io/en/stable/api/training/estimators.html)
+  to be used for executing your code as a SageMaker training job. Our running
+  example is in PyTorch, so we use the `PyTorch` estimator. Except for
+  `entry_point`, arguments depend on the estimator type. Some of the most
+  relevant ones are `instance_type`, `instance_count` (one training job can use
+  several instances), and `role`. The latter is your execution role for the
+  training job: with `get_execution_role()`, you use the default one associated
+  with your AWS account. Finally, `max_run` limits the time of a training job,
+  make sure this is not much shorter than your experiment time.
+  Apart from `sm_estimator`, you can also specify the names of metrics reported
+  to Syne Tune in `metrics_names`, these will be displayed in the dashboard of
+  the training job.
+* If your training script needs extra dependencies not contained in the
+  SageMaker framework, you need to list them in a file `requirements.txt` stored
+  in the same directory as your script. Our running example needs `filelock`,
+  for example.
+
+This launcher script shows how to run training code whose dependencies are
+captured by a SageMaker framework (`PyTorch` in this case), and extra ones
+are specified by a `dependencies.txt` file. However, you may prefer to use a
+Docker image for your training code.
+[launch_height_sagemaker_custom_image.py](examples/launch_height_sagemaker_custom_image.py)
+is an example for how to use such a custom image.
+
+
+## Properties of the SageMaker Back-end
+
+Our running example is not a good candidate to use the SageMaker back-end for.
+It is small, cheap and easy to run, and for such tuning problems, the local
+back-end is the preferred choice. The SageMaker back-end starts a new training
+job for every trial. While this offers great flexibility (you can select a
+SageMaker framework, instance type, instance count), it comes with a substantial
+start-up overhead (often almost two minutes). If training jobs are much longer
+than that, as is often the case with `FIFOScheduler`, these over-heads may not
+matter much.
+
+However, for early stopping schedulers like `HyperbandScheduler`, most trials
+do not run for long, but are stopped or paused after one or a few epochs. For such
+algorithms start-up overheads can matter a lot more. On the other hand, they 
+also profit from increased parallelization.
+
+SageMaker training jobs are not only slow to start, but also have quite some
+overhead for getting stopped. Therefore, if you use ASHA or MOBSTER with the
+SageMaker back-end, we recommend you use `type='promotion'` and specify
```

Comment:
OK, I remove this recommendation, and switch the example to use stopping, so it will work with the current code.

But I really do think that gap should be closed. If we could have a chat on how checkpointing for the SM backend works right now, I can also do the work.

As long as this is not there, we cannot even do comparisons in practice. Note that for the local back-end, the promotion variant often works quite well, and better than stopping towards later, and this is without any overheads.

And I have personally observed stopping an SM training job taking a very long time. That would bother the SM back-end only if such a job would still count as running, so no new trial can be started. Otherwise, it would still be an issue, because the experiment would run > n_workers training jobs at some point in time.